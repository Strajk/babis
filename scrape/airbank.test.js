const fn = require('./airbank');

describe('airbank', () => {
  test('Valid input', () => {
    /* eslint quote-props: ["error", "always"] */
    const INPUT = [
      {
        'Datum provedení': '16/12/2018',
        'Typ úhrady': 'Platba kartou',
        'Částka v měně účtu': '-158',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '-158',
        'Název protistrany': 'DAMEJIDLO .CZ',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '16/12/2018 00:00:00',
        'Referenční číslo': '25725438502',
      },
      {
        'Datum provedení': '16/12/2018',
        'Typ úhrady': 'Příchozí úhrada',
        'Částka v měně účtu': '300',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '300',
        'Název protistrany': 'Spořicí účet',
        'Číslo účtu protistrany': '123456789/3030',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '',
        'Referenční číslo': '25648308902',
      },
      {
        'Datum provedení': '08/12/2018',
        'Typ úhrady': 'Odchozí úhrada',
        'Částka v měně účtu': '-1450',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '-1450',
        'Název protistrany': 'Spořicí účet',
        'Číslo účtu protistrany': '123456789/3030',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '08/12/2018 13:06:24',
        'Referenční číslo': '25501696682',
      },
      {
        'Datum provedení': '04/12/2018',
        'Typ úhrady': 'Příchozí úhrada',
        'Částka v měně účtu': '198',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '198',
        'Název protistrany': 'Joe Doe',
        'Číslo účtu protistrany': '987654321/0600',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': 'Test message',
        'Směnný kurz': '',
        'Datum a čas zadání': '',
        'Referenční číslo': '25421729892',
      },
      {
        'Datum provedení': '03/12/2018',
        'Typ úhrady': 'Odchozí úhrada',
        'Částka v měně účtu': '-1407',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '-1407',
        'Název protistrany': 'John Doe',
        'Číslo účtu protistrany': '987654321/0600',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': 'Test message',
        'Směnný kurz': '',
        'Datum a čas zadání': '03/12/2018 16:24:38',
        'Referenční číslo': '25406315022',
      },
      {
        'Datum provedení': '01/12/2018',
        'Typ úhrady': 'Kreditní úrok',
        'Částka v měně účtu': '39.61',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '39.61',
        'Název protistrany': 'Air Bank',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '',
        'Referenční číslo': '25363022772',
      },
      {
        'Datum provedení': '01/12/2018',
        'Typ úhrady': 'Daň z úroku',
        'Částka v měně účtu': '-5.94',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '-5.94',
        'Název protistrany': 'Air Bank',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '01/12/2018 00:59:55',
        'Referenční číslo': '25363022982',
      },
      {
        'Datum provedení': '24/11/2018',
        'Typ úhrady': 'Platba kartou',
        'Částka v měně účtu': '-376.97',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '-376.97',
        'Název protistrany': 'BILLA, spol. s r. o.',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '',
        'Datum a čas zadání': '24/11/2018 00:00:00',
        'Referenční číslo': '25282317262',
      },
      {
        'Datum provedení': '09/10/2018',
        'Typ úhrady': 'Odměny za placení',
        'Částka v měně účtu': '200',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'CZK',
        'Původní částka úhrady': '200',
        'Název protistrany': 'Air Bank',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': 'Připsání odměn',
        'Směnný kurz': '',
        'Datum a čas zadání': '',
        'Referenční číslo': '24376049442',
      },
      {
        'Datum provedení': '07/10/2018',
        'Typ úhrady': 'Platba kartou',
        'Částka v měně účtu': '-160.25',
        'Poplatek v měně účtu': '',
        'Původní měna úhrady': 'USD',
        'Původní částka úhrady': '-7',
        'Název protistrany': 'GITHUB.COM',
        'Číslo účtu protistrany': '',
        'Poznámka pro mne': '',
        'Zpráva pro příjemce': '',
        'Směnný kurz': '22.89',
        'Datum a čas zadání': '07/10/2018 00:00:00',
        'Referenční číslo': '24393405662',
      },
    ];
    expect(fn.normalize(INPUT)).toMatchSnapshot();
  });
});
