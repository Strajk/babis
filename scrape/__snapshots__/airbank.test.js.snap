// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airbank Valid input 1`] = `
Array [
  Object {
    "amount": 158,
    "date": "2018-12-16",
    "description": "DAMEJIDLO .CZ",
  },
  Object {
    "amount": -300,
    "date": "2018-12-16",
    "description": "Spořicí účet | Příchozí úhrada",
  },
  Object {
    "amount": 1450,
    "date": "2018-12-08",
    "description": "Spořicí účet | Odchozí úhrada",
  },
  Object {
    "amount": -198,
    "date": "2018-12-04",
    "description": "Joe Doe | Příchozí úhrada | Test message",
  },
  Object {
    "amount": 1407,
    "date": "2018-12-03",
    "description": "John Doe | Odchozí úhrada | Test message",
  },
  Object {
    "amount": -39.61,
    "date": "2018-12-01",
    "description": "Air Bank | Kreditní úrok",
  },
  Object {
    "amount": 5.94,
    "date": "2018-12-01",
    "description": "Air Bank | Daň z úroku",
  },
  Object {
    "amount": 376.97,
    "date": "2018-11-24",
    "description": "BILLA, spol. s r. o.",
  },
  Object {
    "amount": -200,
    "date": "2018-10-09",
    "description": "Air Bank | Odměny za placení | Připsání odměn",
  },
  Object {
    "amount": 160.25,
    "date": "2018-10-07",
    "description": "GITHUB.COM (7 USD)",
  },
]
`;
